### ISSUE TITLE
msg: unsupported parameter for module: checksum

### ISSUE THREAD

**Issue by liuzxc** [mitchellh/vagrant#6506]
I was trying to run `vagrant provision`, but got this following error:

```
TASK: [intall nsq ] **************************************************
<127.0.0.1> ESTABLISH CONNECTION FOR USER: vagrant
<127.0.0.1> REMOTE_MODULE get_url url=https://s3.cn-north-1.amazonaws.com.cn/static-files/nsq-0.3.5.linux-amd64.go1.4.2.tar.gz dest=~/nsq-0.3.5.linux-amd64.go1.4.2.tar.gz checksum=SHA256:55c2f30dde00459f4c6215fc2e2e1c66e1b2fa0db7541a419c5a5d0ab7fc1687
<127.0.0.1> EXEC ssh -C -tt -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/Users/liuxingqi/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o Port=2200 -o IdentityFile="/Users/liuxingqi/meiqia/ecoboost/.vagrant/machines/default/virtualbox/private_key" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 127.0.0.1 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765 && echo $HOME/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765'
<127.0.0.1> PUT /var/folders/s9/88z3gh2s6x1gsk9h9114b6f00000gn/T/tmpCUFTkq TO /home/vagrant/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765/get_url
<127.0.0.1> EXEC ssh -C -tt -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/Users/liuxingqi/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o Port=2200 -o IdentityFile="/Users/liuxingqi/meiqia/ecoboost/.vagrant/machines/default/virtualbox/private_key" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 127.0.0.1 /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765/get_url; rm -rf /home/vagrant/.ansible/tmp/ansible-tmp-1447141880.9-149628472858765/ >/dev/null 2>&1'
failed: [default] => {"failed": true}
msg: unsupported parameter for module: checksum

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/liuxin/playbook.retry

default                    : ok=24   changed=0    unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

Any idea what the problem is?

Thanks!

**Comment by gildegoma** [comment]
This is an Ansible error message:

> unsupported parameter for module: checksum

The `checksum` option will be available in Ansible 2.0, but is not present in 1.x versions. See http://docs.ansible.com/ansible/get_url_module.html#options

---

PS: in the future, please use the Ansible or Vagrant mailing list to ask for usage questions. Thanks!

**Comment by liuzxc** [comment]
ok, thanks for help!
